src/pages/edit-product/edit-product.tsx

```diff
@@ -0,0 +1,108 @@
+import { useContext, useRef } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { useParams } from "react-router-dom";
+import ProductForm from "../../components/product-form/product-form";
+import { PRODUCTS_ENDPOINT } from "../../constants";
+import {
+  ProductFormData,
+  ProductFormResolver,
+} from "../../entities/ProductSchema";
+import { useAuth } from "../../hooks/use-auth";
+import { ProductContext } from "../../providers/products-provider";
+import APIClient from "../../services/api-client";
+import { ProductService } from "../../services/product-service";
+
+const EditProduct = () => {
+  const { id } = useParams();
+  const {
+    contextProducts: products,
+    productCategories,
+    refetchProducts,
+  } = useContext(ProductContext);
+  const product = products.find((product) => product.id === id);
+  const formRef = useRef<HTMLFormElement>(null);
+  const formDefaultValues = {
+    name: product?.name || "",
+    category: product?.category?.id || "",
```

Comment:
I don't think that you need "?" after category, because the category of a product cannot be undefined.

Reply:
Indeed, removed "?".